### DISCUSSION TITLE
how to use Vercel GitHub App and GitHub Enterprise with OKTA

### DISCUSSION THREAD

**Discussion by raczosala** [vercel/vercel#5067] [Help] [answered]
While trying to login to github from the vercel dashboard it prompts me to enter my github username and password but we do it through okta, I'm not sure how to connect the vercel github app with an enterprise github repository

Trying to import the repository it will tell me that it cannot be found.

**Comment by styfle** [comment]
~~Vercel doesn't not support GitHub Enterprise at this time. We plan to add this feature in the future, thanks!~~

Updated answer: https://github.com/vercel/vercel/discussions/5067#discussioncomment-1323827

**Comment by av1v3k** [comment]
Now, it's end of 2021, any progress on this @styfle ? Thanks.

**Comment by styfle** [comment, marked as answer]
Vercel now supports login from Okta and other SAML/SSO providers! ([announcement](https://vercel.com/changelog/saml-single-sign-on-and-directory-sync-now-fully-available))

If you're not already on an Enterprise plan, you'll need to [request an upgrade](https://vercel.com/contact/sales).

---

To clarify the [GitHub Enterprise](https://docs.github.com/en/get-started/learning-about-github/githubs-products#github-enterprise) part of this question:

Vercel supports GitHub Enterprise Cloud (cloud-hosted) but not GitHub Enterprise Server (self-hosted).

GitHub Enterprise Server (self-hosted) support is on the roadmap, thanks!

**Reply by jacobhjkim** [reply]
Hi, are there any updates on GitHub Enterprise Server (self-hosted) support? Where can I check the roadmap?

**Reply by simpsoka** [reply]
Hi @jacobhjkim! Thanks for asking. GitHub Enterprise Server is definitely on our radar and we're currently researching it. Currently, we don't have a public roadmap, and I've taken your feedback to the product team that is working on this feature.

**Comment by leerob** [comment]
**Update:** Happy to share it's now much easier to deploy to Vercel with any custom CI/CD provider like GitHub Enterprise Server + GitHub Actions.

[We've created a guide](https://vercel.com/support/articles/how-can-i-use-github-actions-with-vercel) and [an example](https://github.com/vercel/examples/tree/main/ci-cd/github-actions) using GitHub Actions. Would love any feedback!

**Reply by jacobhjkim** [reply]
Hi @leerob, thanks for the GitHub Actions example. Sadly, the provided example does not run with Github Actions on Enterprise. Vercel CLI tries to log in with GitHub, but it fails since the GitHub Enterprise credential does not match the Vercel account. 
```shell
$ vercel pull --yes --environment=production --token=${{ secrets.VERCEL_TOKEN }}

Vercel CLI 27.3.4
Loading scopes…
Searching for existing projects…
> You must re-authenticate to use {YOUR_WORKSPACE} scope.
> Log in to Vercel 
● Continue with GitHub 
○ Continue with GitLab 
○ Continue with Bitbucket 
○ Continue with Email 
  ───────────────────────
○ Abort 
Error: Process completed with exit code [13].
```

Since Vercel OAuth does not support GitHub Enterprise, should we re-create our accounts with an email authentication method?

**Reply by robinyoong** [reply]
Hi @jacobhjkim! Thanks for sharing this with us. This issue may not be specifically related to Github Enterprise Server. Can you check if your `--token` is scoped to the correct team, and that has [not expired](https://vercel.com/changelog/expiration-dates-now-available-for-access-tokens)?

There's also an [option to select "No Expiration"](https://user-images.githubusercontent.com/47657586/184083158-53479925-a78d-4164-ada9-a071f99d893b.png) for cases of CI.

**Reply by jacobhjkim** [reply]
Sorry, it was indeed an issue with my vercel token. Issue resolved 👍
